release-testing/store-component-versions.go

```diff
@@ -0,0 +1,45 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/cyberark/conjur-oss-suite-release/pkg/repositories"
+)
+
+func main() {
+	var suiteFile string
+	flag.StringVar(&suiteFile, "f", "", "Repository YAML file to parse")
+	flag.Parse()
+
+	if suiteFile == "" {
+		flag.PrintDefaults()
+		log.Fatal("missing required -f flag")
+		return
+	}
+
+	repoConfig, err := repositories.NewConfig(suiteFile)
+	if err != nil {
+		log.Fatal(err)
+		return
+	}
+
+	for _, categories := range repoConfig.Section.Categories {
```

Comment:
not a blocker at all but we should probably have an abstraction layer method for grabbing the versions so that we don't have to embed an external struct's layout in here.

Reply:
can you elaborate on this one please?

Reply:
The code you have here is intricately tied 1:1 to the exact internal structure of `Repository` struct so it's tightly coupled with the code you are calling (and as such fragile). If you have a helper method to give you this list of versions that you call here but who's logic lives within `repository` package then the coupling is low and logic that could break would live close to the code that changes (making this logic here more resilient).